daisy/error.go

```diff
@@ -65,39 +67,57 @@ func addErrs(e dErr, errs ...error) dErr {
 	return e
 }
 
-func errf(format string, a ...interface{}) dErr {
-	return newErr(fmt.Errorf(format, a...))
+// Errf returns a DError by constructing error message with given format.
+func Errf(format string, a ...interface{}) DError {
+	return newErr(format, fmt.Errorf(format, a...))
 }
 
-// newErr returns a dErr. newErr is used to wrap another error as a dErr.
-// If e is already a dErr, e is copied and returned.
+// newErr returns a DError. newErr is used to wrap another error as a DError.
+// If e is already a DError, e is copied and returned.
+// If e is a normal error, replacementMsg is used to hide privacy info.
 // If e is nil, nil is returned.
-func newErr(e error) dErr {
+func newErr(replacementMsg string, e error) DError {
 	if e == nil {
 		return nil
 	}
 	if dE, ok := e.(*dErrImpl); ok {
 		return dE
 	}
-	return &dErrImpl{errs: []error{e}}
+	return &dErrImpl{errs: []error{e}, errFormats: []string{replacementMsg}}
```

Comment:
errFormat and replacementMsg naming is not in sync. Both are also not very descriptive of what they actually hold (anonymized error message). Consider renaming both to something like anonymizedErrs or similar. It's already called safeErrMsg in some pieces of code. That one is better although it's not clear what the message is safe from.  anonymized described exactly what these messages are

Reply:
Renamed  to anonymizedErrs/anonymizedErrMsg as suggested.